instrumentation/mongo/src/test/java/brave/mongo/ITTraceMongoCommandListener.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2013-2020 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.mongo;
+
+import brave.ScopedSpan;
+import brave.Tracing;
+import brave.propagation.ThreadLocalCurrentTraceContext;
+import brave.sampler.Sampler;
+import com.mongodb.MongoClientSettings;
+import com.mongodb.MongoQueryException;
+import com.mongodb.ServerAddress;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.event.CommandListener;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import zipkin2.Span;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.assertj.core.api.Assertions.entry;
+import static org.junit.Assume.assumeTrue;
+
+public class ITTraceMongoCommandListener {
+  private static final String A_COLLECTION_NAME = "myCollection"; // collection doesn't have to exist
+  private static final String INVALID_COLLECTION_NAME = "?.$";
+
+  private final List<Span> spans = new CopyOnWriteArrayList<>();
+
+  private final Tracing tracing = tracingBuilder(Sampler.ALWAYS_SAMPLE).build();
+  private MongoClient mongoClient;
+  private MongoDatabase database;
+
+  @Before public void init() {
+    final String databaseName = System.getenv("MONGODB_DB");
```

Comment:
if we test with system managed resource, travis should start it and set the variables needed.. you can look for our mysql stuff for example. We don't want any tests that aren't actually run in CI

Reply:
Can you point me to where MySQL is configured? I saw the environment variables in the MySQL tests but I don't see where those are set up, or how MySQL is being requested on Travis.

I could also attempt to use https://github.com/flapdoodle-oss/de.flapdoodle.embed.mongo to start a MongoDB from the test itself.

Reply:
maybe best to just use testcontainers instead. it should be easy to validate this is working in travis. We use testcontainers in many projects (zipkin, zipkin-reporter)

https://github.com/testcontainers/testcontainers-java/tree/master/examples/mongodb-container/src/test/java/org/testcontainers/containers